### PR TITLE
Simplified WriteAsync code

### LINKED ISSUE #2088: Check if we streamline WriteAsync
WriteAsync declares a local function to avoid using async/await due to a mono bug that existed years ago. Hopefully, that's fixed, so we can streamline the method.

https://github.com/realm/realm-dotnet/blob/6228b00777793780a24e0348c0d6ad6cc85cc43e/Realm/Realm/Realm.cs#L712-L743

### PR DESCRIPTION
**Pull request by papafe**
## Description
Simplifies code for `WriteAsync` methods, using `async/await`. This was not done in the past due to a mono bug.

https://github.com/realm/realm-dotnet/blob/6228b00777793780a24e0348c0d6ad6cc85cc43e/Realm/Realm/Realm.cs#L712-L743

Fixes #2088

### CODE REVIEW

File: Realm/Realm/Realm.cs [hunk 1]
```diff
@@ -748,11 +748,11 @@ public T Write<T>(Func<T> function)
         /// Action to execute inside a <see cref="Transaction"/>, creating, updating, or removing objects.
         /// </param>
         /// <returns>An awaitable <see cref="Task"/>.</returns>
-        public Task WriteAsync(Action<Realm> action)
+        public async Task WriteAsync(Action<Realm> action)
         {
             Argument.NotNull(action, nameof(action));
 
-            return WriteAsync(tempRealm =>
+            await WriteAsync(tempRealm =>
```
Comment (nirinchev) [inline Realm/Realm/Realm.cs:755]: I don't think we need to change this. Not that it's that big of a deal but there's no need to allocate a state machine for this method.

**Review by nirinchev — APPROVED** [review]
Nice! Glad to see it has been fixed. As you probably guessed, for PRs like this, there's no need for a changelog entry as it doesn't do anything user-facing.

**Comment by papafe** [comment]
Corrected according to comment, the CI results are fine (apart from the usual flaky tests)

**Comment by nirinchev** [comment]
Go for it

**Pull request merged**